test/runtests.jl

```diff
@@ -194,6 +194,24 @@ end
 end
 
 
+@testset "shift operations" begin
+    vals = [rand(Int8, 8)..., rand(Int64, 8)..., rand(Int1024, 8)...]
+    shifts = [rand(-8:8, 8)..., rand(-64:64, 8)..., rand(-1024:1024, 8)...]
```

Comment:
While you are at it, would you also test for non-Int shifts ? e.g.
 ```suggestion
    shifts = [rand(-8:8, 8)..., rand(-64:64, 8)..., rand(Int256(-1024):Int256(1024), 8)...]
```